src/Illuminate/Notifications/Events/BroadcastNotificationCreated.php

```diff
@@ -77,8 +77,10 @@ public function broadcastWith()
      */
     protected function channelName()
     {
-        $class = str_replace('\\', '.', get_class($this->notifiable));
-
-        return $class.'.'.$this->notifiable->getKey();
+        if (method_exists($this->notifiable, 'getNotificationChannel')) {
+            return $this->notifiable->getNotificationChannel();
+        } else {
```

Comment:
Don't worry about StyleCI, a fix PR will be done after.
I think the most important is that there are tests as @ntzm mentionned